Yoast/javascript
# Extract replacement variable editor to a new package

## Summary
This PR moves the ReplacementVariableEditor to a new package so it is easier to use it in other packages.

The `@yoast/search-metadata-previews` has been kept backwards-compatible so that dependent packages keep working.

In the future, we might remove the exports from `@yoast/search-metadata-previews` so it might be wise to update references from `@yoast/search-metadata-previews` to `@yoast/replacement-variable-editor`.


This PR can be summarized in the following changelog entry:

* [@yoast/replacement-variable-editor] Introduce a new package for the replacement variable editor.


## Test instructions

This PR can be tested by following these steps:
1. Checkout branch and link the branch in Plugin.
1. Build everything 🎉 
1. Check the Search preview, behaviour and looks should be equal to the live version.
1. Check the Search appearance, behaviour and looks should be equal to the live version.
1. Do a short smoketest through the plugin, check that nothing breaks.
1. Try to run the `example-app` using `yarn install && yarn start` in the `apps/components` directory. Make sure it runs and do a smoketest there as well.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
	* Search appearance
	* Google search preview 

## UI changes

## Quality assurance


Fixes #505
